jme3-core/src/main/java/com/jme3/renderer/RenderManager.java

```diff
@@ -117,6 +127,89 @@ public class RenderManager {
     public RenderManager(Renderer renderer) {
         this.renderer = renderer;
         this.forcedOverrides.add(boundDrawBufferId);
+        // register default pipeline context
+        contexts.put(PipelineContext.class, new DefaultPipelineContext());
+    }
+    
+    /**
+     * Gets the default pipeline used when a ViewPort does not have a
+     * pipeline already assigned to it.
+     * 
+     * @return 
+     */
+    public RenderPipeline getPipeline() {
+        return defaultPipeline;
+    }
+    
+    /**
+     * Sets the default pipeline used when a ViewPort does not have a
+     * pipeline already assigned to it.
+     * <p>
+     * default={@link ForwardPipeline}
+     * 
+     * @param pipeline default pipeline (not null)
+     */
+    public void setPipeline(RenderPipeline pipeline) {
+        this.defaultPipeline = pipeline;
+    }
+    
+    /**
+     * Gets the default pipeline context registered under
+     * {@link PipelineContext#getClass()}.
+     * 
+     * @return 
+     */
+    public PipelineContext getDefaultContext() {
+        return getContext(PipelineContext.class);
+    }
+    
+    /**
+     * Gets the pipeline context registered under the class.
+     * 
+     * @param <T>
+     * @param type
+     * @return 
+     */
+    public <T extends PipelineContext> T getContext(Class<T> type) {
+        return (T)contexts.get(type);
+    }
+    
+    /**
+     * Gets the pipeline context registered under the class or creates
+     * and registers a new context supplied by the supplier.
+     * 
+     * @param <T>
+     * @param type
+     * @param supplier
+     * @return 
+     */
+    public <T extends PipelineContext> T getOrCreateContext(Class<T> type, Supplier<T> supplier) {
+        T c = getContext(type);
+        if (c == null) {
+            c = supplier.get();
+            contexts.put(type, c);

```

Comment:
registerContext only puts the context into the map, i would call that function instead of putting it into the map inside getOrCreate too, in case some changes are made later on there is only a single place where the values are added.